plan.go

```diff
@@ -40,17 +46,23 @@ func (p *plan) Replace(ctx context.Context, from interface{}, to interface{}) er
 	}
 
 	for i := range p.order {
-		b := p.order[i]
-		if f.Name == b.Name {
-			// same function, lets replace it
-			p.order[i] = t
-			return nil
+		for j := range p.order[i] {
+			b := p.order[i][j]
+			if f.Name == b.Name {
+				// same function, lets replace it
+				p.order[i][j] = t
+				return nil
+			}
 		}
 	}
 	return errors.New("builder not found")
 }
 
 func (p *plan) Run(ctx context.Context, initData ...interface{}) (Result, error) {
+	return p.RunParallel(ctx, 1, initData...)
+}
+
+func (p *plan) RunParallel(ctx context.Context, workers uint, initData ...interface{}) (Result, error) {
```

Comment:
How feasible would it be to provide the user with a maxWorker option? ie. after resolving the dependencies can data-builder see the maximum amout of parallelism possible and create this many workers?

Reply:
thats an interesting question, Normally I would suggest number of workers to be proportional to cpu cores/threads.
spawning more workers than number of cores/threads may not always be helpful, specially for cpu intensive tasks.

I can add an option to read max parallelism from a given plan, what interface will you prefer ?

Reply:
I see, having the user restrict the number of workers makes sense in a cpu-intensive scenario. but in my use case, I know network calls will be the bottleneck. should we replace the `workers` arg with `maxWorkers`? the actual number of workers used could be `min(plan.maxParallelism(), maxWorkers)`. eg. suppose resolve dependencies returns:
```
{{"Name3"}, {"Name2", "Name5"}, {"Name1", "Name4"}}
```
in this case `plan.maxParallelism()` = 2 as we know there won't be any major advantage to setting workers > 2

Reply:
yes, this should be simple enough to implement, do you think maybe I can create a static function in data-builder, something along the lines of
```go
func MaxPlanParallelism(p Plan) uint
{
    ....
}
```

Reply:
I have added this function